static/assets/js/entity-reference/app.js

```diff
@@ -53,30 +63,44 @@
                             return entity.name == name;
                         });
                     }
-                    return entity;
+                    return !entity.details['x-hidden'];
                 })
-                .map(setDisplayTypesOnEntity);
+                .map(function(entity) 
+                { 
+                    return setDisplayTypesOnEntity(entity, swagger.definitions); 
+                });
         }
 
-        function setDisplayTypesOnEntity(entity) {
+        function setDisplayTypesOnEntity(entity, definitions) {
             Object.keys(entity.details.properties).forEach(function(propertyName) {
                 var property = entity.details.properties[propertyName];
+                property.isArray = (property.type === "array");
 
-                if (property.type == "array") {
-                    var ref = property.items.$ref.replace("#/definitions/", "");
-                    property.displayType = ref;
-                    property.ref = ref;
-                    property.isArrayRef = true;
-                } else if (property.$ref != undefined) {
-                    var ref = property.$ref.replace("#/definitions/", "");
-                    property.displayType = ref;
-                    property.ref = ref;
-                    property.isSingleRef = true;
+                if (property.isArray) {
+                    property.ref = (property.items.$ref && property.items.$ref.replace("#/definitions/", ""));
+                    property.displayType = getRefDisplayName(property.ref, definitions) || getTypeDisplayName(property.items);
                 } else {
-                    property.displayType = property.type;
-                }
+                    property.ref = property.$ref && property.$ref.replace("#/definitions/", "");
+                    property.displayType = getRefDisplayName(property.ref, definitions) || getTypeDisplayName(property);
+                } 
+
+                property.descriptionHtml = $sce.trustAsHtml(property.description);
             });
             return entity;
         }
+
+        function getRefDisplayName(ref, definitions) {
+            if (ref) {
+                return (definitions[ref]['x-display-name'] || ref).toLowerCase();
+            }
+        }
+
+        function getTypeDisplayName(property) {
```

Comment:
Thinking out loud here... 
I kind of wanted to make it more generic than just looking for 'date-time'. If we add more, then doing `if (format == x) else if (format === y) else if ...` will be harder to read and less performant (albeit only slightly). I can move the table down into the function to make it closer to its usage for readability, but this will cause the array to be created on every scope entrance rather than when the class was created. Probably a good trade-off, though for readability...